include/ensmallen_bits/delta_bar_delta/update_policies/delta_bar_delta_update.hpp

```diff
@@ -0,0 +1,201 @@
+  //! Modify the initialStepSize hyperparameter.
+  double& InitialStepSize() { return initialStepSize; }
+
+  //! Access the kappa hyperparameter.
+  double Kappa() const { return kappa; }
+  //! Modify the kappa hyperparameter.
+  double& Kappa() { return kappa; }
+
+  //! Access the phi hyperparameter.
+  double Phi() const { return phi; }
+  //! Modify the phi hyperparameter.
+  double& Phi() { return phi; }
+
+  //! Access the theta hyperparameter.
+  double Theta() const { return theta; }
+  //! Modify the theta hyperparameter.
+  double& Theta() { return theta; }
+
+  //! Access the minStepSize hyperparameter.
+  double MinStepSize() const { return minStepSize; }
+  //! Modify the minStepSize hyperparameter.
+  double& MinStepSize() { return minStepSize; }
+
+  /**
+   * The UpdatePolicyType policy classes must contain an internal 'Policy'
+   * template class with two template arguments: MatType and GradType. This is
+   * instantiated at the start of the optimization, and holds parameters
+   * specific to an individual optimization.
+   */
+  template <typename MatType, typename GradType>
+  class Policy
+  {
+   public:
+    typedef typename MatType::elem_type ElemType;
+
+    /**
+     * This is called by the optimizer method before the start of the iteration
+     * update process.
+     *
+     * @param parent Instantiated parent class.
+     * @param rows Number of rows in the gradient matrix.
+     * @param cols Number of columns in the gradient matrix.
+     */
+    Policy(
+        const DeltaBarDeltaUpdate& parent,
+        const size_t rows,
+        const size_t cols) :
+        parent(parent),
+        kappa(ElemType(parent.kappa)),
+        phi(ElemType(parent.phi)),
+        theta(ElemType(parent.theta)),
+        minStepSize(ElemType(parent.minStepSize))
+    {
+      delta_bar.zeros(rows, cols);
+      epsilon.set_size(rows, cols);
+      epsilon.fill(parent.InitialStepSize());
+    }
+
+    /**
+     * Update step for Gradient Descent.
+     *
+     * @param iterate Parameters that minimize the function.
+     * @param stepSize Step size to be used for the given iteration.
+     * @param delta The gradient matrix.
+     */
+    void Update(MatType& iterate,
+                const double stepSize,
+                const GradType& delta)
+    {
+      const MatType signMatrix = sign(delta % delta_bar);
+      const MatType incrementMask = conv_to<MatType>::from(signMatrix == +1);
+      const MatType decrementMask = conv_to<MatType>::from(signMatrix == -1);
+
+      epsilon += incrementMask * kappa;
+      epsilon -= decrementMask * phi % epsilon;
+      epsilon.clamp(minStepSize,
+          arma::Datum<typename MatType::elem_type>::inf);
+
+      delta_bar *= theta;
+      delta_bar += (1 - theta) * delta;
+
+      iterate -= epsilon % delta;
+    }
+
+   private:
+    //! The instantiated parent class.
+    const DeltaBarDeltaUpdate& parent;
+
+    //! The exponential average of past gradients.
+    MatType delta_bar;
```

Comment:
```suggestion
    MatType deltaBar;
```
Just to match the style guide, we use camel-case vs. snake-case.